Reflection_Engine/Modify/SetPropertyValue.cs

```diff
@@ -107,18 +107,15 @@ private static bool SetValue(this IBHoMObject obj, string propName, object value
         {
             if (obj == null) return false;
 
-            if (!obj.CustomData.ContainsKey(propName))
+            if (value is IFragment)
             {
-                if (value is IFragment)
-                {
-                    obj.Fragments.Add(value as IFragment);
-                    Compute.RecordWarning("The objects does not contain any property with the name " + propName + ". The value is being set as a fragment.");
-                }
-                else
-                {
-                    obj.CustomData[propName] = value;
-                    Compute.RecordWarning("The objects does not contain any property with the name " + propName + ". The value is being set as custom data.");
-                }
+                obj.Fragments.Add(value as IFragment);
+                Compute.RecordWarning("The objects does not contain any property with the name " + propName + ". The value is being set as a fragment.");
```

Comment:
```suggestion
                Compute.RecordWarning("The object does not contain any property with the name " + propName + ". The value is being set as a fragment.");
```

Or

```suggestion
                Compute.RecordWarning("The objects do not contain any property with the name " + propName + ". The value is being set as a fragment.");
```

`objects does` doesn't read quite right to me, one of them should be singular but not sure what the rest of the method is doing so suggesting both options. Looks like the method takes in a single object so the first one might be more appropriate?